src/test/java/org/openrewrite/java/spring/framework/HttpComponentsClientHttpRequestFactoryReadTimeoutTest.java

```diff
@@ -0,0 +1,110 @@
+
+                      return new RestTemplateBuilder()
+                              .requestFactory(() -> {
+                                  HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
+                                  clientHttpRequestFactory.setReadTimeout(30000);
+                                  // ... set poolingConnectionManager on HttpClient
+                                  return clientHttpRequestFactory;
+                              })
+                              .build();
+                  }
+              }
+              """,
+            """
+              import org.apache.hc.core5.http.config.Registry;
+              import org.apache.hc.core5.http.config.RegistryBuilder;
+              import org.apache.hc.core5.http.io.SocketConfig;
+              import org.apache.hc.client5.http.socket.ConnectionSocketFactory;
+              import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager;
+              import org.springframework.boot.web.client.RestTemplateBuilder;
+              import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+              import org.springframework.web.client.RestTemplate;
+
+              import java.util.concurrent.TimeUnit;
+
+              class RestContextInitializer {
+                  RestTemplate getRestTemplate() throws Exception {
+                      Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create().build();
+                      PoolingHttpClientConnectionManager poolingConnectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+                      poolingConnectionManager.setDefaultSocketConfig(SocketConfig.custom().setSoTimeout(30000, TimeUnit.MILLISECONDS).build());
+
+                      return new RestTemplateBuilder()
+                              .requestFactory(() -> {
+                                  HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
+                                  // ... set poolingConnectionManager on HttpClient
+                                  return clientHttpRequestFactory;
+                              })
+                              .build();
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    // TODO Additional scenarios not yet covered
+    // - Using BasicHttpClientConnectionManager
+    // - Using PoolingHttpClientConnectionManagerBuilder
+    // - No HttpClientConnectionManager at all
+    // - No intermediate variable for connectionManager
+    // - setReadTimeout called with local variable not accessible near connection manager
+    // - there already is a setDefaultSocketConfig call
```

Comment:
As indicated here the recipe right now is very specific to a particular test scenario; there's various others that we might want to cover as well, and a few that are quite hard to cover well. Let's do a best effort pass through these to see which we can still reasonably implement, and then merge this first iteration.